### ISSUE TITLE
add_to_minv: obj not free (3, 53, 98) - when adherer is attacked with polearm

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#227]
Reported by mobileuser.

Likely caused by b1b092f89

**Comment by elunna** [comment]
Other items that could be affected:
* Aklys: Not a problem. Just sticks to the adherher.
* Chakrams: Also not a problem, the autoreturn doesn't seem to have problems and just sticks to the adherer.
* Lance: Also a problem, we get the same error.

**Comment by elunna** [comment]
```
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7362859 in __GI_abort () at abort.c:79
#2  0x0000555555a7f0c1 in NH_abort () at end.c:236
#3  0x0000555555a860b5 in panic (str=0x555556598ee0 "add_to_minv: obj not free (%d, %d, %d)")
    at end.c:802
#4  0x0000555555da914b in add_to_minv (mon=0x6110000025c0,
    obj=<error reading variable: Cannot access memory at address 0xffffffffffffffc3>)
    at mkobj.c:2472
#5  0x00005555561e3384 in mpickobj (mtmp=0x6110000025c0, otmp=0x60c0000004c0) at steal.c:572
#6  0x0000555555a22352 in thitmonst (mon=0x6110000025c0, obj=0x60c0000004c0) at dothrow.c:2200
#7  0x00005555557dd9c5 in use_pole (obj=0x60c0000004c0, autohit=0 '\000') at apply.c:4308
#8  0x00005555557e5bf3 in doapply () at apply.c:5190
#9  0x0000555555899ced in rhack (cmd=0x5555568a0bc0 <in_line> "a") at cmd.c:5756
#10 0x000055555578b0d0 in moveloop (resuming=0 '\000') at allmain.c:870
#11 0x000055555646f3cc in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
```

**Comment by elunna** [comment]
Fixed in 8a8feb8c9
